In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [30]:
# lectura de datos de tablas fct
fct_precios_insiders_AAPL = pd.read_csv(('../Dataset/clean/AAPL_fct_precios_insider.csv'))
fct_precios_insiders_AAPL.head()

Unnamed: 0  trade_date ticker      insider_name title trade_type  \
0           0  2015-01-23   AAPL  Riccio Daniel J.   SVP          S   
1           1  2015-02-18   AAPL       Jung Andrea   Dir          S   
2           2  2015-03-06   AAPL      Maestri Luca   CFO          S   
3           3  2015-03-09   AAPL      Maestri Luca   CFO          S   
4           4  2015-03-18   AAPL      Maestri Luca   CFO          S   

   quantity_of_shares  owned delta_owned    value   Open  Close  movimiento  
0                3804      0       -100%   428955  24.98  25.13        0.15  
1               40000  14595        -73%  5125200  28.50  28.75        0.25  
2                3400  14124        -19%   437920  28.68  28.27       -0.41  
3                2800  11324        -20%   361116  28.58  28.39       -0.19  
4               10823    501        -96%  1394219  28.36  28.69        0.33

In [34]:
# eliminar el simbolo % y el guion - de la columna "delta_owned".
fct_precios_insiders_AAPL['delta_owned'] = fct_precios_insiders_AAPL['delta_owned'].str.replace('%', '').str.replace('-', '')

In [35]:
fct_precios_insiders_AAPL

Unnamed: 0  trade_date ticker        insider_name title trade_type  \
0             0  2015-01-23   AAPL    Riccio Daniel J.   SVP          S   
1             1  2015-02-18   AAPL         Jung Andrea   Dir          S   
2             2  2015-03-06   AAPL        Maestri Luca   CFO          S   
3             3  2015-03-09   AAPL        Maestri Luca   CFO          S   
4             4  2015-03-18   AAPL        Maestri Luca   CFO          S   
..          ...         ...    ...                 ...   ...        ...   
216         216  2024-10-04   AAPL        Maestri Luca   CFO          S   
217         217  2024-11-15   AAPL   Levinson Arthur D   Dir          S   
218         218  2024-11-18   AAPL         Kondo Chris   PAO          S   
219         219  2024-12-16   AAPL  Williams Jeffrey E   COO          S   
220         220  2025-02-03   AAPL   Levinson Arthur D   Dir          S   

     quantity_of_shares    owned delta_owned     value    Open   Close  \
0                  3804        0         100    428955   24.98   25.13   
1                 40000    14595          73   5125200   28.50   28.75   
2                  3400    14124          19    437920   28.68   28.27   
3                  2800    11324          20    361116   28.58   28.39   
4                 10823      501          96   1394219   28.36   28.69   
..                  ...      ...         ...       ...     ...     ...   
216               59305   107788          35  13433769  227.40  226.30   
217              200000  4215576           5  45464500  226.15  224.75   
218                4130    15419          21    945233  225.00  227.77   
219              100000   389944          20  24997395  247.72  250.76   
220                1516  4215576           0    343147  229.74  227.76   

     movimiento  
0          0.15  
1          0.25  
2         -0.41  
3         -0.19  
4          0.33  
..          ...  
216       -1.10  
217       -1.40  
218        2.77  
219        3.04  
220       -1.98  

[221 rows x 13 columns]

# COMENTARIO
Necesitaba quitar el símbolo de % de la columna delta_owned porque quiero realizar una métrica para saber si el insider vendió una parte significativa de sus acciones. Por ejemplo, lo quiero estimar en un umbral del 50%.

Hay que entender que la columna delta_owned representa el % de acciones que tiene esa persona luego de haber realizado una transacción, por ejemplo, si es venta el delta_owned será una disminución, si es compra será un aumento. 

Para el caso de las personas que su delta_owned es 0, no quiere decir que no hicieron transacciones, quiere decir que el porcentaje negociado es ínfimo frente al total de las acciones que poseen. Esto se puede evidenciar en la línea 220 de la tabla fct.

In [36]:
# transformar la columna "delta_owned" a tipo int 
fct_precios_insiders_AAPL['delta_owned'] = fct_precios_insiders_AAPL['delta_owned'].astype(int)

In [37]:
fct_precios_insiders_AAPL

Unnamed: 0  trade_date ticker        insider_name title trade_type  \
0             0  2015-01-23   AAPL    Riccio Daniel J.   SVP          S   
1             1  2015-02-18   AAPL         Jung Andrea   Dir          S   
2             2  2015-03-06   AAPL        Maestri Luca   CFO          S   
3             3  2015-03-09   AAPL        Maestri Luca   CFO          S   
4             4  2015-03-18   AAPL        Maestri Luca   CFO          S   
..          ...         ...    ...                 ...   ...        ...   
216         216  2024-10-04   AAPL        Maestri Luca   CFO          S   
217         217  2024-11-15   AAPL   Levinson Arthur D   Dir          S   
218         218  2024-11-18   AAPL         Kondo Chris   PAO          S   
219         219  2024-12-16   AAPL  Williams Jeffrey E   COO          S   
220         220  2025-02-03   AAPL   Levinson Arthur D   Dir          S   

     quantity_of_shares    owned  delta_owned     value    Open   Close  \
0                  3804        0          100    428955   24.98   25.13   
1                 40000    14595           73   5125200   28.50   28.75   
2                  3400    14124           19    437920   28.68   28.27   
3                  2800    11324           20    361116   28.58   28.39   
4                 10823      501           96   1394219   28.36   28.69   
..                  ...      ...          ...       ...     ...     ...   
216               59305   107788           35  13433769  227.40  226.30   
217              200000  4215576            5  45464500  226.15  224.75   
218                4130    15419           21    945233  225.00  227.77   
219              100000   389944           20  24997395  247.72  250.76   
220                1516  4215576            0    343147  229.74  227.76   

     movimiento  
0          0.15  
1          0.25  
2         -0.41  
3         -0.19  
4          0.33  
..          ...  
216       -1.10  
217       -1.40  
218        2.77  
219        3.04  
220       -1.98  

[221 rows x 13 columns]

In [40]:
# crear una nueva columna para determinar si el insider ha vendido o comprado en base al valor de la columna "delta_owned" mayor al 50% debe colocar 1 es venta significativa y 0 si no lo es

fct_precios_insiders_AAPL['significant_transaction'] = np.where(fct_precios_insiders_AAPL['delta_owned'] > 60, 1, 0)
fct_precios_insiders_AAPL.head(10)

Unnamed: 0  trade_date ticker       insider_name title trade_type  \
0           0  2015-01-23   AAPL   Riccio Daniel J.   SVP          S   
1           1  2015-02-18   AAPL        Jung Andrea   Dir          S   
2           2  2015-03-06   AAPL       Maestri Luca   CFO          S   
3           3  2015-03-09   AAPL       Maestri Luca   CFO          S   
4           4  2015-03-18   AAPL       Maestri Luca   CFO          S   
5           5  2015-04-02   AAPL  Ahrendts Angela J   SVP          S   
6           6  2015-04-06   AAPL  Ahrendts Angela J   SVP          S   
7           7  2015-04-16   AAPL       Maestri Luca   CFO          S   
8           8  2015-04-20   AAPL       Maestri Luca   CFO          S   
9           9  2015-04-30   AAPL   Riccio Daniel J.   SVP          S   

   quantity_of_shares  owned  delta_owned    value   Open  Close  movimiento  \
0                3804      0          100   428955  24.98  25.13        0.15   
1               40000  14595           73  5125200  28.50  28.75        0.25   
2                3400  14124           19   437920  28.68  28.27       -0.41   
3                2800  11324           20   361116  28.58  28.39       -0.19   
4               10823    501           96  1394219  28.36  28.69        0.33   
5               25000  99728           20  3119874  27.92  27.99        0.07   
6               44197  55531           44  5607762  27.80  28.44        0.64   
7                 700   5936           11    88788  28.20  28.18       -0.02   
8                5936      0          100   758080  28.04  28.50        0.46   
9               24090  72255           25  3031463  28.73  27.95       -0.78   

   significant_transaction  
0                        1  
1                        1  
2                        0  
3                        0  
4                        1  
5                        0  
6                        0  
7                        0  
8                        1  
9                        0